lib/methods/platform/waitForStateTransitionResultFactory.js

```diff
@@ -0,0 +1,69 @@
+const {
+  v0: {
+    PlatformPromiseClient,
+    WaitForStateTransitionResultRequest,
+  },
+} = require('@dashevo/dapi-grpc');
+
+/**
+ *
+ * @param {GrpcTransport} grpcTransport
+ * @returns {waitForStateTransitionResult}
+ */
+function waitForStateTransitionResultFactory(grpcTransport) {
+  /**
+   * @typedef waitForStateTransitionResult
+   * @param {Buffer} stateTransitionHash
+   * @param {boolean} prove
+   * @param {DAPIClientOptions & getDocumentsOptions} [options]
+   * @returns {Promise<Object>}
+   */
+  async function waitForStateTransitionResult(stateTransitionHash, prove, options = {}) {
+    // eslint-disable-next-line no-param-reassign
+    options = {
+      // Set default timeout
+      ...options,
+      timeout: 60000,
```

Comment:
I guess we should allow devs to overwrite timeout so I would move it before applying options.